### PR TITLE
Fix crash that would happen when renaming in mixed f#/c# project.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/28474

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @jinujoseph @jasonmalinowski

File: src/EditorFeatures/Test2/Rename/InlineRenameTests.vb [hunk 1]
```diff
@@ -1965,5 +1965,46 @@ End Class
                 Assert.Equal(InlineRenameFileRenameInfo.NotAllowed, session.FileRenameInfo)
             End Using
         End Sub
+
+        <WpfFact>
+        <WorkItem(28474, "https://github.com/dotnet/roslyn/issues/28474")>
+        <Trait(Traits.Feature, Traits.Features.Rename)>
+        Public Async Function HandleProjectsWithoutCompilations() As Task
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test2/Rename/InlineRenameTests.vb]: validated test failure/crash prior to fix.

File: src/EditorFeatures/Test2/Rename/InlineRenameTests.vb [hunk 2]
```diff
@@ -1965,5 +1965,46 @@ End Class
...
+                            <ProjectReference>CSharpProject</ProjectReference>
+                            <Document>
+                                // a no-compilation document
+                            </Document>
+                        </Project>
+                    </Workspace>)
+
+                Dim session = StartSession(workspace)
+
+                ' Type a bit in the file
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Test2/Rename/InlineRenameTests.vb]: Why does this need a fully blown test with the editors? If you look in https://github.com/dotnet/roslyn/blob/master/src/EditorFeatures/Test2/Rename/RenameEngineTests.vb there's a bunch of tests that are just "rename Goo to Bar" tests are the usual case. This file is normally intended for stuff specific to the inline rename experience.

As a comparison, think of this as being like the completion command handler tests: sometimes you need to test that, but if you're just trying to assert that an item is shown you test the completion provider directly.

File: src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs [hunk 3]
```diff
@@ -522,7 +525,7 @@ bool metadataTypeMatches(SymbolAndProjectIdSet s, INamedTypeSymbol t)
             //          /
             //         └
             //        E
-            // and we're passed in 'B, C, E' as hte project to search, then this set 
+            // and we're passed in 'B, C, E' as the project to search, then this set
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs]: What have you done with the real @CyrusNajmabadi?

File: src/EditorFeatures/Test2/Rename/InlineRenameTests.vb (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Test2/Rename/InlineRenameTests.vb]: don't know, don't care :)  i just wanted a test that validates the case :)   i didn't have a lot of time to do this PR, and i was sad there were customers complaining about hte scenario.

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski want to merge in?  You're also welcome to take on the PR and update the test if you want :)

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski want to merge in? You're also welcome to take on the PR and update the test if you want :)

**Comment by jasonmalinowski** [comment]
@CyrusNajmabadi Yeah I'll give it an update. Just got back from a trip and so still trying to catch up on some stuff.

**Comment by TIHan** [comment]
Thank you @CyrusNajmabadi for this.

File: src/EditorFeatures/Test2/Rename/InlineRenameTests.vb (hunk 2 above)
Comment (jasonmalinowski) [reply src/EditorFeatures/Test2/Rename/InlineRenameTests.vb]: Pushed the conversion over to your branch. Will merge once CI runs.

Comment (CyrusNajmabadi) [reply src/EditorFeatures/Test2/Rename/InlineRenameTests.vb]: > Pushed the conversion over to your branch. Will merge once CI runs.

Thanks!  IMO, we should keep both tests.  There's enough interesting logic in both places that i think it's worth covering our bases.  Up to you!

**Comment by jasonmalinowski** [comment]
Thanks @CyrusNajmabadi for the fix!

**Pull request merged**